### ISSUE TITLE
Add synaptic connections for neurons from Jovanovic larval paper

### ISSUE THREAD

**Issue by dosumis** [FlyBase/drosophila-anatomy-developmental-ontology#245]
Spreadsheet of synaptic connections here:
https://ars.els-cdn.com/content/image/1-s2.0-S0092867416312429-mmc5.csv

 - [x] Generate a two column tsv mapping neuron names used in header and column 1 of this table to FBbt ids.  Ignore individual names & collapse l/r distinction.  e.g.  A09a a1l Basin-2 | A09a   a1r Basin-2  => FBbt id for  Basin 2 . @Clare72 

- [x] Use Python Pandas to generate a [robot template](http://robot.obolibrary.org/template)

Sketch of Robot template: First row => column header names for reference.  Second row = template.  Subsequent rows = variable fillers (binding to % in template).  

| ID | Type | Basin-2 | Basin-1 | Basin-4 | ...| 
|--|--|--|--|--|--|
| ID | CLASS_TYPE | C 'synapsed to' % | C 'synapsed to' % | C 'synapsed to '% | ...|
|  FBbt:12345678 | subclass | FBbt:87654321 | FBbt:65432187 |... |
...

(To check: is synapsed_to the correct direction here?)

There should be a template column for each distinct neuron type (collapsing left/right distinction) .  Cells can be left blank where there is no synaptic connection

Spec for transformation:  Iterate over input table - adding neuron type mapped from the first column to the ID column of the template.   Only add a neuron to column named for it if synaptic connections are present according to input table. (Question - do we only record if left and right present?)

[ipython Jupyter Notebook](https://github.com/FlyBase/drosophila-anatomy-developmental-ontology/blob/jovanovic-neurons/fbbt/src/patterns/robot_template_projects/Jovanovic_circuit/Jovanovic_circuit_template_gen.py.ipynb) to roll [Jovanovic.owl](https://github.com/FlyBase/drosophila-anatomy-developmental-ontology/blob/Jovanovic.owl)

- [x] Run robot template command; check output; run robot merge:

`robot template --input fbbt-edit.owl --template <template file name> --output <output file name>`

**Comment by dosumis** [comment]
Due to a robot bug running into quirks of relations in fbbt-edit.obo, we need to run generation like this:

`robot template --input-iri http://purl.obolibrary.org/obo/ro.owl --template Jovanovic_template.csv --output Jovanovic.owl`

If the file looks OK, it can be merged with the editor's file: 

`robot merge --input fbbt-edit.owl --input Jovanovic_template.owl --output fu.owl`

**Comment by dosumis** [comment]
@Clare72 - can you finish the neuron mapping?  You could actually do this inside the notebook extending: 

```python
lookup = { 'A09a a1l Basin-2': 'FBbt:00111229',
           'A09a a1r Basin-2': 'FBbt:00111229',
           'A09b a1l Basin-1': 'FBbt:00111227',
           'A09b a1r Basin-1': 'FBbt:00111227',
           'A09c a1l Basin-4': 'FBbt:00111230',
           'A09c a1r Basin-4': 'FBbt:00111230'}  # ... This should be completed for all columns
```

**Comment by dosumis** [comment]
@Clare72 - We also need to discuss interpretation of the original table, esp re: direction synaptic connections.

**Comment by ah963** [comment]
Note: this is related to [#188](https://github.com/FlyBase/drosophila-anatomy-developmental-ontology/issues/188) and to [VFB#140](https://github.com/VirtualFlyBrain/Management/issues/140) and [VFB#33](https://github.com/VirtualFlyBrain/Management/issues/33)

**Comment by Clare72** [comment]
>Data S2. Matrix of Synaptic Connectivity, Related to Figure 2. Numbers reflect the number of synapses between neurons, from the neuron labeled for the column to the neuron labeled for the row.

**Comment by Clare72** [comment]
I am having some trouble knowing what to do with the lch/vch neurons.
They have split lch5 into four groups (1, 2/4, 3 and 5), but only one group is in the ontology (lch5) - shall I collapse all to lch5 or should I make the subgroups in the ontology?
Also, lch1 has a subclass for abdominal lch1 - but looks like all lch1 are abdominal, so this seems redundant... Which FBbt ID should I put in the table?
Same issue for vch2.
vch1 has the same abdominal subclass, but also has throracic subclasses - so I guess I use the FBbt for the abdominal class here?
'vch neuron' is a superclass for vch1+vch2 (vchA/B in paper), but subclasses of vch are not superclasses of vch1+2 subclasses (I think I should change this..?).
Also v'ch is a synonym for lch1 (used in the paper), should we be more careful about v'ch/vch, or do you think it is clear enough?

**Comment by dosumis** [comment]
Will try to take a look today.  Quick answer: I used this site as a guide for nomenclature and relationships abdominal sense organs and sensory neurons.
http://www.normalesup.org/~vorgogoz/FlyPNS/PNSnomenclature0.html

Does this help?

(Also recall someone in VFB (Alex?) recently posting a new paper on larval PNS/sensory system but can't seem to find where.)

**Comment by Clare72** [comment]
Thanks, this is a useful reference for synonyms (I think all the relevant ones are already in the ontology - but nice to see it all in a table), but it doesn't really help with the issue of which FBbt classes should represent the neurons in the paper.

**Comment by Clare72** [comment]
I made a provisional table of neurons and their fbbt terms and ran the script. May need to edit this later (see earlier comment).

**Comment by dosumis** [comment]
Not sure I follow what the problem is with vch neurons.  The only have one neuron each and classification looks OK to me:

![image](https://user-images.githubusercontent.com/112839/41549558-2c0e6134-731e-11e8-848c-cb8f8f538864.png)

**Comment by dosumis** [comment]
BTW - think it makes sense to always use the general abdominal class as the connectomic data is from A1.

**Comment by Clare72** [comment]
The problem is that 'abdominal vch1 neuron' and 'abdominal vch2 neuron' are not 'abdominal vch neuron's and I think the term I need is the latter, as they are grouped as vchA/B in the paper. I can change this easily.

**Comment by dosumis** [comment]
OK, must have missed that. Some simple logical definitions with part of could fix the classification.

**Comment by Clare72** [comment]
Directionality seems to be wrong here - Basin-2 is synapsed_to Basin-1 (should be other way around).
Should I change the template to synapsed_by, transpose the input connectivity table, or do both (independently) and generate both sets of relationships?

**Comment by Clare72** [comment]
Did both in the end - also corrected an issue (in two places) in the template generator where entries were being overwritten with blanks if earlier-read cells were >0, but later-read cells were ==0 for same pairing.
I didn't check the merge exhaustively (would take all afternoon!), but the few classes I did check looked perfect for synapsed_by and _to compared to the spreadsheet (including those previously affected by above issue).
I also removed pre-existing assertions of connectivity between this set of neurons before merging - looks like these were based on the same dataset with less detail? (Ohyama et al., 2015).

Now going to remove interconnectivity assertions from parent classes and add FBrf.
